shared/fs/header/add-new-container.js

```diff
@@ -4,14 +4,49 @@ import * as FsGen from '../../actions/fs-gen'
 import flags from '../../util/feature-flags'
 import {compose, setDisplayName, connect, type Dispatch, type TypedState} from '../../util/container'
 import AddNew from './add-new'
+import {isDarwin, isMobile, isIOS} from '../../constants/platform'
 
 const mapStateToProps = (state: TypedState) => ({})
 
 const mapDispatchToProps = (dispatch: Dispatch, {routePath}) => ({
   _newFolderRow: (parentPath: Types.Path) => dispatch(FsGen.createNewFolderRow({parentPath})),
+  _upload: (parentPath: Types.Path, type: Types.OpenDialogType) =>
+    dispatch(FsGen.createPickAndUpload({parentPath, type})),
 })
 
-const mergeProps = (stateProps, {_newFolderRow}, {path, style}) => {
+const desktopUploadItems = (path: Types.Path, _upload) =>
+  isDarwin
+    ? [
+        {
+          onClick: () => _upload(path, 'both'),
+          icon: 'iconfont-upload',
+          title: 'Upload a file or folder',
+        },
+      ]
+    : [
+        // Linux/Windows don't support accepting file and dir at the same time.
```

Comment:
What does this mean?

Reply:
You can have a popup modal dialog to choose either a file, or a folder, on macOS. On Linux/Windows, you have to decide whether you want a file or a folder, and that popup is just for that.

Reply:
Hmmm, that's pretty terrible UX. I'd almost rather just do files then. Do people often want to do whole-folder actions on mobile?

Reply:
I feel like it can be useful, and it's in the [design](https://app.zeplin.io/project/56b25ffc64fc91773a73aaf8/screen/5a7212277692923af00a8836). Would it be less confusing if we just split them into separate items on all platform?

Reply:
That's the desktop design, and it's definitely useful there. I'm questioning whether there's any point to it on mobile.

Reply:
Oh I missed the "on mobile" part. This is just for the desktop. The mobile ones are from `mobileUploadItems` below.

Reply:
Oh right. Ugh, I guess we don't have a choice then :(